### PR TITLE
[Enctools] Macro centralization

### PR DESCRIPTION
**Pull request by lwan108**
Signed-off-by: LijuanWang <lijuan.wang@intel.com>

### CODE REVIEW

**Review by takulako — APPROVED** [review]

File: _studio/shared/include/mfx_config_encode.h [hunk 1]
```diff
@@ -26,7 +26,6 @@
 #if defined(MFX_ENABLE_H264_VIDEO_ENCODE)
     #undef MFX_ENABLE_H264_PRIVATE_CTRL
     #define MFX_ENABLE_APQ_LQ
-    #define MFX_ENABLE_H264_REPARTITION_CHECK
```
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/shared/include/mfx_config_encode.h:29]: This is windows only macro, so removed it from linux.

[the remark this answers is not in the document]
Comment (dmitryermilov) [reply _studio/shared/include/mfx_config_encode.h:29]: This is a **wrong** assumption: https://github.com/Intel-Media-SDK/MediaSDK/blob/2bbad9fbdc980f3b698d4fc3fb679e5885adeb30/_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp#L592

[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/shared/include/mfx_config_encode.h:29]: Windows only was suggested in PR#[2180](https://github.com/intel-innersource/drivers.gpu.media.sdk.lib/pull/2180).
Can we get MFX_ENABLE_H264_REPARTITION_CHECK clarified offline and submit a separated after confirmation due to MFX_ENABLE_H264_REPARTITION_CHECK code has been trimmed already?
So we can get this PR merged.

[the remark this answers is not in the document]
Comment (dmitryermilov) [reply _studio/shared/include/mfx_config_encode.h:29]: It's fine to me. Created https://github.com/oneapi-src/oneVPL-intel-gpu/issues/173 to track

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**